pkg/kotsadmsnapshot/backup.go

```diff
@@ -930,3 +946,59 @@ func prepareIncludedNamespaces(namespaces []string) []string {
 	}
 	return includedNamespaces
 }
+
+// excludeShutdownPodsFromBackup will exclude pods that are in a shutdown state from the backup
+// this is to prevent the hook backup from failing if a pod is in a shutdown state and cannot be backed up
+func excludeShutdownPodsFromBackup(ctx context.Context, clientset kubernetes.Interface, backup *velerov1.Backup) (err error) {
+	namespaces := backup.Spec.IncludedNamespaces
+	if len(namespaces) == 0 {
+		return
+	}
+
+	// if the backup includes all namespaces, we need to get the list of namespaces
+	if namespaces[0] == "*" {
+		namespaces, err = getNamespaces(ctx, clientset)
+		if err != nil {
+			return errors.Wrap(err, "failed to get namespaces")
+		}
+	}
+
+	for _, namespace := range namespaces {
+		pods, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+			FieldSelector: "status.phase=Failed",
```

Comment:
to scope this even further, i would suggest adding a `kots.io/backup: velero` label selector as well (also has constants `kotsadmtypes.BackupLabel` and `kotsadmtypes.BackupLabelValue` defined)

Reply:
in fact, there should be two selections, one with `kots.io/backup: velero` and `kots.io/kotsadm: true`, and the other with `kots.io/backup: velero` and `kots.io/app-slug: <app-slug>` so that only the relevant pods get modified.